src/org/opendatakit/briefcase/ui/pull/components/PullFormsTableView.java

```diff
@@ -0,0 +1,113 @@
+
+public class PullFormsTableView extends JTable {
+  static final String[] HEADERS = new String[]{"", "Form Name", "Pull Status", ""};
+  static final Class[] TYPES = new Class[]{Boolean.class, String.class, String.class, JButton.class};
+  static final boolean[] EDITABLE_COLS = new boolean[]{true, false, false, false};
+
+  static final int SELECTED_CHECKBOX_COL = 0;
+  static final int FORM_NAME_COL = 1;
+  static final int PULL_STATUS_COL = 2;
+  static final int DETAIL_BUTTON_COL = 3;
+
+  public PullFormsTableView(PullFormsTableViewModel model) {
+    super(model);
+    setName("forms");
+
+    addMouseListener(new MouseAdapterBuilder().onClick(this::relayClickToButton).build());
+
+    Dimension formNameDims = getHeaderDimension(HEADERS[FORM_NAME_COL]);
+    Dimension pullStatusDims = getHeaderDimension(HEADERS[PULL_STATUS_COL]);
+
+    setRowHeight(28);
+
+    TableColumnModel columns = getColumnModel();
+    columns.getColumn(SELECTED_CHECKBOX_COL).setMinWidth(40);
+    columns.getColumn(SELECTED_CHECKBOX_COL).setMaxWidth(40);
+    columns.getColumn(SELECTED_CHECKBOX_COL).setPreferredWidth(40);
+    columns.getColumn(FORM_NAME_COL).setMinWidth(formNameDims.width + 25);
+    columns.getColumn(FORM_NAME_COL).setPreferredWidth(formNameDims.width + 25);
+    columns.getColumn(PULL_STATUS_COL).setMinWidth(pullStatusDims.width + 25);
+    columns.getColumn(PULL_STATUS_COL).setPreferredWidth(pullStatusDims.width + 25);
+    columns.getColumn(DETAIL_BUTTON_COL).setCellRenderer(cellWithButton());
+    columns.getColumn(DETAIL_BUTTON_COL).setMinWidth(40);
+    columns.getColumn(DETAIL_BUTTON_COL).setMaxWidth(40);
+    columns.getColumn(DETAIL_BUTTON_COL).setPreferredWidth(40);
+
+    setFillsViewportHeight(true);
+
+    TableRowSorter<PullFormsTableViewModel> sorter = sortBy(getModel(), FORM_NAME_COL, ASCENDING);
+    setRowSorter(sorter);
+    sorter.sort();
+  }
+
+  private Dimension getHeaderDimension(String header) {
+    return getTableHeader()
+        .getDefaultRenderer()
+        .getTableCellRendererComponent(null, header, false, false, 0, 0)
+        .getPreferredSize();
+  }
+
+
+  private void relayClickToButton(MouseEvent event) {
+    int column = getColumnModel().getColumnIndexAtX(event.getX());
+    int row = event.getY() / getRowHeight();
+
+    if (row < getRowCount() && row >= 0 && column < getColumnCount() && column >= 0) {
+      Object value = getValueAt(row, column);
+      if (value instanceof JButton)
+        ((JButton) value).doClick();
+    }
+  }
+
+  @Override
+  public PullFormsTableViewModel getModel() {
+    return (PullFormsTableViewModel) super.getModel();
+  }
+
+  private static TableCellRenderer cellWithButton() {
+    return (table, value, isSelected, hasFocus, row, column) -> {
+      JButton button = (JButton) value;
+      button.setOpaque(true);
+      button.setBackground(isSelected ? table.getSelectionBackground() : table.getBackground());
+      return button;
+    };
+  }
```

Comment:
I’d like your opinion on this alternative. Is it simpler or clearer?

![screen shot 2018-05-22 at 9 34 08 pm](https://user-images.githubusercontent.com/55099/40403671-27a4864e-5e08-11e8-9387-1e5e935bd62e.png)

Reply:
That is much much much better! Thanks